compiler/luci-interpreter/src/Interpreter.cpp

```diff
@@ -171,31 +192,30 @@ class HookImpl final : public EventNotifier
     assert(tensor != nullptr);
     for (const auto &observer : _observers)
     {
-      observer->postTensorWrite(_loader_map.tensor_to_node.at(tensor), tensor);
+      observer->postTensorWrite(_runtime_to_ir.tensor_to_node.at(tensor), tensor);
     }
   }
 
 private:
-  RuntimeToIR &_loader_map;
+  RuntimeToIR &_runtime_to_ir;
   const std::vector<ExecutionObserver *> &_observers;
 };
 
 Interpreter::Interpreter(const luci::Module *module)
 {
   _runtime_to_ir = std::make_unique<RuntimeToIR>();
-  _event_notifier = std::make_unique<HookImpl>(*_runtime_to_ir, _observers);
+  _event_notifier = std::make_unique<EventNotifierImpl>(*_runtime_to_ir, _observers);
```

Comment:
If runtime module is used in the `interpret` method, `Interpreter` may not need `_event_notifier`.

Reply:
`Interpreter` is the owner of `EventNotifierImpl` instance, and `RuntimeModule` accesses it via `EventNotifier` interface.
`EventNotifierImpl` cannot be moved into `RuntimeModule`, because `EventNotifierImpl` depends on IR, while `core` component must not.
(If I understood you correctly.)

Reply:
> EventNotifierImpl cannot be moved into RuntimeModule, because EventNotifierImpl depends on IR, while core component must not.

I'm not sure I understood the answer. CMakeLists.txt in the `src/core` directory links luci_lang (this is IR, isn't it?) to luci_interpreter_core (it is used in KernelParams.h).

You may have some design principles for luci-interpreter, but it is difficult for me to understand them without explanation (I'm not very familiar with cpp software designs). Can you explain a little bit more about the restrictions or principles of luci-interpreter? Even a brief explanation would be greatly helpful for me to review future PRs.